src/ctapipe/io/dl2_tables_preprocessing.py

```diff
@@ -0,0 +1,476 @@
+"""Module containing classes related to event loading and preprocessing"""
+
+from pathlib import Path
+
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import AltAz, SkyCoord
+from astropy.table import Column, QTable, Table, vstack
+
+try:
+    from pyirf.simulations import SimulatedEventsInfo
+    from pyirf.spectral import (
+        DIFFUSE_FLUX_UNIT,
+        POINT_SOURCE_FLUX_UNIT,
+        PowerLaw,
+        calculate_event_weights,
+    )
+    from pyirf.utils import calculate_source_fov_offset, calculate_theta
+except ModuleNotFoundError:
```

Comment:
I think with this many imports, it's simpler to do something like this:

```
try:
    from ...
    has_pyirf = True
except ModuleNotFounderror:
    has_pyirf = False
```

and then raise on `has_pyirf = False`.


```